allennlp/training/metrics/conll_coref_scores.py

```diff
@@ -227,7 +227,7 @@ def phi4(gold_clustering, predicted_clustering):
         """
         return (
             2
-            * len([mention for mention in gold_clustering if mention in predicted_clustering])
```

Comment:
This one I would probably keep, for readability.  Does just using `len()` without an internal list not work?

Reply:
```python
len(mention for mention in gold_clustering if mention in predicted_clustering)
```

When you omit the parentheses and have only one argument that is a comprehension expression, it's a generator (with more arguments you need the parentheses). A generator doesn't define `__len__`, so it'd crash.

Reply:
Yeah, I guess python doesn't want to silently try to call `len()` on an infinite generator.  Fair enough.  Yeah, I'd keep this list.

Reply:
I also now removed a `float` conversion there that is unnecessary in Python 3.

Reply:
FWIW, I like to use `sum(1 for x in y)` in these cases, but keeping the list might be more straightforward.